src/lsm/scan_builder.zig

```diff
@@ -46,43 +46,38 @@ pub fn ScanBuilderType(
         const ScanBuilder = @This();
 
         pub const Scan = ScanType(Groove, Storage);
+        const Slots = std.ArrayListUnmanaged(Scan);
 
-        // Since `ScanTree` consume memory and IO, it is limited by `lsm_scans_max`,
-        // however, merging scans does not require resources.
-        // E.g, if `lsm_scans_max = 4` we can have at most 3 merge operations:
-        // M₁(M₂(C₁, C₂), M₃(C₃, C₄))
-        const ScanSlots = stdx.BoundedArray(Scan, constants.lsm_scans_max);
-        const MergeSlots = stdx.BoundedArray(Scan, constants.lsm_scans_max - 1);
-
-        scan_slots: *ScanSlots,
-        merge_slots: *MergeSlots,
+        scan_slots: Slots,
+        merge_slots: Slots,
```

Comment:
Again, would be great to "why" in the commit message!

And, as usual, every time I see bounded array I am tempted by something simple like

```
scans: *[constants.lsm_scans_max]Scan = undefined;
scan_count: u32 = 0;
```